runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/grpc/GrpcWindmillStreamFactory.java

```diff
@@ -209,11 +261,29 @@ public GetWorkerMetadataStream createGetWorkerMetadataStream(
         onNewWindmillEndpoints);
   }
 
-  private StreamObserverFactory newStreamObserverFactory() {
+  private StreamObserverFactory newBufferringStreamObserverFactory() {
```

Comment:
this is confusing name since it doesn't buffer beyond what grpc stream internally does. It just respects grpc flow control which it seems like we'd want to do in the direct path case as well.

There is a different BufferingStreamObserver for fnapi which does do buffering with a queue.

Reply:
reverted